### ISSUE TITLE
After hook crashes if log files exist

### ISSUE THREAD

**Issue by csvan** [archfz/cypress-terminal-report#57]
The following code in BaseOutputProcessor.js:

``` 
hasSpecChunkWritten(spec) {
    return !!this.specChunksWritten[spec];
}
```

will throw if a log file for the current spec already exists in the output folder. This is because the following doesn't get invoked:

```
if (!fs.existsSync(this.file)) {
  this.prepareForWrite();
}
```

Due to this, `this.specChunksWritten` is never initialized before `hasSpecChunkWritten` tries to access it.

**Comment by archfz** [comment]
Do you have steps to reproduce so we can write a test for this case?

**Comment by csvan** [comment]
1. Add a standard Cypress setup with at least one spec file
2. Add to plugins:

```
  require('cypress-terminal-report/src/installLogsPrinter')(on, {
    // defaultTrimLength: "500",
    // commandTrimLength: "500",
    outputRoot: 'cypress/',
    specRoot: 'cypress/integration',
    printLogsToFile: 'always', // <--- so that we always try to access the file
    printLogsToConsole: 'never',
    outputTarget: {
      'logs|txt': 'txt',
      'logs|json': 'json',
    },
  });
```

Add to support:

```
require('cypress-terminal-report/src/installLogsCollector')();
```

2. add <name of spec>.txt to cypress/logs

3. run run cypress

**Comment by archfz** [comment]
Released in 2.3.1
